cartographer/mapping/map_builder_interface.h

```diff
@@ -90,6 +90,11 @@ class MapBuilderInterface {
 
   virtual const std::vector<proto::TrajectoryBuilderOptionsWithSensorIds>&
   GetAllTrajectoryBuilderOptions() const = 0;
+
+  // Sets the callback function that is invoked whenever the global optimization
+  // problem is solved.
+  virtual void SetGlobalSlamOptimizationCallback(
```

Comment:
I'd move the function to PoseGraphInterface, this would be a simpler solution.
Users can always call MapBuilderInterface::pose_graph()::SetGlobalSlamOptimizationCallback.